asg/src/reducer/monoid/tuple.rs

```diff
@@ -25,18 +25,29 @@ macro_rules! tuple_monoid_impls {
         $(
             impl<$($T:Monoid),+> Monoid for ($($T,)+) {
 
-                fn append(self, other: ($($T,)+)) -> ($($T,)+) {
+                fn append(self, other: Self) -> Self {
                     ($(self.$idx.append(other.$idx)),+,)
                 }
 
-                fn append_all(self, others: impl Iterator<Item = ($($T,)+)>) -> ($($T,)+) {
+                fn append_all(self, others: impl Iterator<Item = ($($T,)+)>) -> Self {
                     others.fold(self, |acc, tup| acc.append(tup))
                 }
             }
         )+
     }
 }
 
+impl Monoid for () {
+    fn append(self, _: Self) -> Self {
+        ()
+    }
+
+    fn append_all(self, _: impl Iterator<Item=Self>) -> Self {
+        ()
+    }
+}
```

Comment:
Hmm, this should not be necessary? The macro should be able to expand to this using `Tuple0 {}`?

Reply:
It seems to be necessary since
- A comma is necessary for a unary tuple. 
- An expression is expected if a comma is in an empty tuple.